File: dataset/ZeroShot - Copy/CLAUDE_claude-sonnet-4-5-20250929/Task191.c

#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int max(int a, int b) {
    return a > b ? a : b;
}

int longestValidParentheses(char* s) {
    if (s == NULL || strlen(s) == 0) {
        return 0;
    }
    
    int len = strlen(s);
    int maxLen = 0;
    int* dp = (int*)calloc(len, sizeof(int));
    
    for (int i = 1; i < len; i++) {
        if (s[i] == ')') {
            if (s[i - 1] == '(') {
                dp[i] = (i >= 2 ? dp[i - 2] : 0) + 2;
            } else if (i - dp[i - 1] > 0 && s[i - dp[i - 1] - 1] == '(') {
                dp[i] = dp[i - 1] + 2 + (i - dp[i - 1] >= 2 ? dp[i - dp[i - 1] - 2] : 0);
            }
            maxLen = max(maxLen, dp[i]);
        }
    }
    
    free(dp);
    return maxLen;
}

int main() {
    // Test case 1
    printf("Test 1: %d\\n", longestValidParentheses("(()"));
    
    // Test case 2
    printf("Test 2: %d\\n", longestValidParentheses(")()())"));
    
    // Test case 3
    printf("Test 3: %d\\n", longestValidParentheses(""));
    
    // Test case 4
    printf("Test 4: %d\\n", longestValidParentheses("()(())"));
    
    // Test case 5
    printf("Test 5: %d\\n", longestValidParentheses("(()())"));
    
    return 0;
}

File: dataset/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Task191.py

def longest_valid_parentheses(s):
    if not s:
        return 0
    
    max_len = 0
    dp = [0] * len(s)
    
    for i in range(1, len(s)):
        if s[i] == ')':
            if s[i - 1] == '(':
                dp[i] = (dp[i - 2] if i >= 2 else 0) + 2
            elif i - dp[i - 1] > 0 and s[i - dp[i - 1] - 1] == '(':
                dp[i] = dp[i - 1] + 2 + (dp[i - dp[i - 1] - 2] if i - dp[i - 1] >= 2 else 0)
            max_len = max(max_len, dp[i])
    
    return max_len

if __name__ == "__main__":
    # Test case 1
    print("Test 1:", longest_valid_parentheses("(()"))
    
    # Test case 2
    print("Test 2:", longest_valid_parentheses(")()())"))
    
    # Test case 3
    print("Test 3:", longest_valid_parentheses(""))
    
    # Test case 4
    print("Test 4:", longest_valid_parentheses("()(()"))
    
    # Test case 5
    print("Test 5:", longest_valid_parentheses("(()())"))
